sdk/cosmos/examples/get_database.rs

```diff
@@ -20,7 +21,9 @@ async fn main() -> Result<(), Box<dyn Error + Send + Sync>> {
 
     let database_client = client.into_database_client(database_name.clone());
 
-    let response = database_client.get_database().execute().await?;
+    let response = database_client
+        .get_database(Context::new(), GetDatabaseOptions::new())
```

Comment:
Some options won't be able to have default implementations because they have required arguments. If we opt for `Default`, we won't be able to do that for _all_ option types. 

Unfortunately since Rust doesn't have default params, this is fairly verbose and `Default` impls don't help necessarily. We could have one method `get_database` that only takes a context and default the options, and a `get_database_with_options` which takes an options param.

Reply:
Generally, if a parameter is required (hence making it required to create options) it's its own parameter. Eventually we'll want to be consistent with the other SDKs when not idiomatic e.g. which parameters are parameters and which are in some sort of options bag.